tracking/R3BTrackingS515.cxx

```diff
@@ -0,0 +1,559 @@
+        hist[i] = new TH1F(hname, hname,100,min_offset[i],max_offset[i]);
+    }
+    std::cout << "\n\n-- Perfroming minimization for the detector alignment. Please wait... \n";
+
+    // Setting up Minimizer function parameters 
+    Double_t precision = 1e-10;//0 - default precision will be automaticalle determined
+    Double_t tolerance = 0.02;
+    minimizer->SetMaxFunctionCalls(1000000000); // for Minuit/Minuit2
+    minimizer->SetMaxIterations(100);  // for GSL
+    minimizer->SetTolerance(tolerance);
+    minimizer->SetPrecision(precision);
+    minimizer->SetPrintLevel(2);
+    minimizer->SetStrategy(0);//to run faster
+    ROOT::Math::Functor f(&AlignmentErrorS515, NVarsFunctor);
+    minimizer->SetFunction(f);
+    minimizer->SetPrintLevel(0);
+
+    // Getting experimental data and doing minimization 
+    UInt_t i;
+    Int_t bs=0;
+    bool good_minimum = false;
+    while(bs<100)//running several minimizations
+    {
+        minimizer->Clear();
+        for(i=0; i<NVarsFunctor; i++)//sampling +=50% from limits
+        {
+            offset[i] = gRandom->Uniform(min_offset[i]*0.5, max_offset[i]*0.5);
+            minimizer->SetVariable(i,Form("par%d",i),offset[i], step[i]);
+            minimizer->SetVariableLimits(i, min_offset[i], max_offset[i]);	
+        }
+        minimizer->Minimize();
+        xs = minimizer->X();
+        //if(minimizer->Status() !=0) continue; //valid minimum
+        //Check if all paramters are "far" from limits
+        for(i=0; i<NVarsFunctor; i++)
+        { 
+            if(fabs(  (xs[i]-min_offset[i]) / min_offset[i] )<0.1 ||
+                    fabs( (max_offset[i]-xs[i]) / max_offset[i] )<0.1    )
+                break;
+        }
+        if(i!=NVarsFunctor)
+        {   
+            //std::cout << "\n\n -- Parameter is too close to the limit!! ";
+            continue;
+        }
+        else
+        {
+            std::cout << "\n\n-- Good parameters are found!!";
+            good_minimum = true;
+        }
+        std::cout << "\n-- Minimization No. " << bs << "\n-- Minimum: f(";
+        for(i=0; i<NVarsFunctor; i++)
+        { 
+            std::cout << xs[i]; 
+            hist[i]->Fill(xs[i]);
+            if(i!=NVarsFunctor-1)
+                std::cout << ",";
+        }
+        std::cout << ") = " << minimizer->MinValue();
+        std::cout << "\n-- Minimizer status: " << minimizer->Status() << std::endl;
+        bs++;
+    }
+    // Outputing final info and histograms
+    if ( minimizer->MinValue()  < tolerance  && f(xs) < tolerance)
+        std::cout << "-- Minimizer " << "   converged to the right minimum" << std::endl;
+    else
+    {
+        std::cout << "-- Minimizer "  << "   failed to converge !!!" << std::endl;
+    }
+    TCanvas * c1 = new TCanvas("c1","c1",1000,1000);
+    c1->Divide(5,4);
+    for(UInt_t v=0; v<NVarsFunctor; v++)
+    {
+        c1->cd(v+1);
+        hist[v]->Draw();
+    }
+    return;
+}
+
+double AlignmentErrorS515(const double * par)
```

Comment:
double R3BTrackingS515::AlignmentErrorS515(const double * par)

Reply:
I have changed this function to be a static method of the class (see my last commit). The reason to declare it as a global friend function is because it has to be accessed by an external class ROOT::Math::Functor. Both methods (friend or static) seem to work fine but I agree that it looks better as a proper member of the class